Sources/Amplitude/Plugins/iOS/UIKitUserInteractions.swift

```diff
@@ -0,0 +1,153 @@
+
+    private static let initializeSwizzle: () = {
+        swizzleSendAction()
+    }()
+
+    static func register(_ amplitude: Amplitude) {
+        queue.sync {
+            amplitudeInstances.add(amplitude)
+        }
+        initializeSwizzle
+    }
+
+    private static func swizzleSendAction() {
+        let applicationClass = UIApplication.self
+
+        let originalSelector = #selector(UIApplication.sendAction)
+        let swizzledSelector = #selector(UIApplication.amp_sendAction)
+
+        guard
+            let originalMethod = class_getInstanceMethod(applicationClass, originalSelector),
+            let swizzledMethod = class_getInstanceMethod(applicationClass, swizzledSelector)
+        else { return }
+
+        let originalImp = method_getImplementation(originalMethod)
+        let swizzledImp = method_getImplementation(swizzledMethod)
+
+        class_replaceMethod(applicationClass,
+            swizzledSelector,
+            originalImp,
+            method_getTypeEncoding(originalMethod))
+        class_replaceMethod(applicationClass,
+            originalSelector,
+            swizzledImp,
+            method_getTypeEncoding(swizzledMethod))
+    }
+}
+
+extension UIApplication {
+    private var keyWindow: UIWindow? {
+        connectedScenes
+            .compactMap { $0 as? UIWindowScene }
+            .flatMap { $0.windows }
+            .first { $0.isKeyWindow }
+    }
+
+    @objc dynamic func amp_sendAction(_ action: Selector, to target: Any?, from sender: Any?, for event: UIEvent?) -> Bool {
+        let sendActionResult = amp_sendAction(action, to: target, from: sender, for: event)
+
+        guard
+            sendActionResult,
+            let keyWindow = keyWindow,
+            let view = sender as? UIView
+        else { return sendActionResult }
+
+        if let textField = view as? UITextField, !textField.shouldTrack(action, for: event) {
+            return sendActionResult
+        } else {
+            #if !os(tvOS)
+            if let slider = view as? UISlider, !slider.shouldTrack(action, for: event) {
+                return sendActionResult
+            }
+            #endif
+        }
+
+        let viewData = view.extractData(with: action, in: keyWindow)
+
+        UIKitUserInteractions.amplitudeInstances.allObjects.forEach {
+            $0.track(event: UserInteractionEvent(
+                viewController: viewData.viewController,
+                title: viewData.title,
+                accessibilityLabel: viewData.accessibilityLabel,
+                actionMethod: viewData.actionMethod,
+                targetViewClass: viewData.targetViewClass,
+                targetText: viewData.targetText,
+                hierarchy: viewData.hierarchy))
+        }
+
+        return sendActionResult
+    }
+}
+
+extension UIView {
+    private static let viewHierarchyDelimiter = " -> "
+
+    struct ViewData {
```

Comment:
How consistent (in terms of property structure) will ViewData be when adding support for additional view types? 

For every addition, you'd have to modify ViewData and the UserInteractionEvent initializer, it may be easier to just return the properties dict directly.

Reply:
TargetText is the only property that is specific to UIButtons, while the other properties are general for all other view types. I think it's better to send the properties one by one to the UserInteractionEvent initializer to be specific about what properties we're sending along with the event.

As we implement the fallback for SwiftUI support and other views that do not trigger action methods, we might add a couple more properties, but the current property structure is quite general, IMO.